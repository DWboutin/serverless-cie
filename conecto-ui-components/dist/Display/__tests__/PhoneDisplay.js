"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _Display = require("../../Display");

var _Logo = _interopRequireDefault(require("../../Logo"));

var _Button = _interopRequireDefault(require("../../Button"));

var _reactIntl = require("react-intl");

var _SVG = require("../../SVG");

global.console = {
  error: jest.fn()
};
describe('PhoneDisplay', () => {
  it('should match snapshot', () => {
    const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Display.PhoneDisplay, {
      logo: _react.default.createElement(_Logo.default, {
        type: "roofing",
        suffix: _react.default.createElement(_reactIntl.FormattedMessage, {
          id: "contractor"
        })
      }),
      button: _react.default.createElement(_Button.default, {
        type: "bordered"
      }, "question@contect.ca"),
      bottomSvgComponent: _react.default.createElement(_SVG.RoofingPhoneBg, null)
    }, "test"));
    expect(wrapper).toMatchSnapshot();
  });
  it('should console.error when not passing a logo', () => {
    (0, _enzyme.shallow)(_react.default.createElement(_Display.PhoneDisplay, {
      button: _react.default.createElement(_Button.default, {
        type: "bordered"
      }, "question@contect.ca"),
      bottomSvgComponent: _react.default.createElement(_SVG.RoofingPhoneBg, null)
    }, "test"));
    expect(global.console.error).toBeCalled();
  });
  it('should console.error when not passing a button', () => {
    (0, _enzyme.shallow)(_react.default.createElement(_Display.PhoneDisplay, {
      logo: _react.default.createElement(_Logo.default, {
        type: "roofing",
        suffix: _react.default.createElement(_reactIntl.FormattedMessage, {
          id: "contractor"
        })
      }),
      bottomSvgComponent: _react.default.createElement(_SVG.RoofingPhoneBg, null)
    }, "test"));
    expect(global.console.error).toBeCalled();
  });
  it('should console.error when not passing a bottomSvgComponent', () => {
    (0, _enzyme.shallow)(_react.default.createElement(_Display.PhoneDisplay, {
      logo: _react.default.createElement(_Logo.default, {
        type: "roofing",
        suffix: _react.default.createElement(_reactIntl.FormattedMessage, {
          id: "contractor"
        })
      }),
      button: _react.default.createElement(_Button.default, {
        type: "bordered"
      }, "question@contect.ca")
    }, "test"));
    expect(global.console.error).toBeCalled();
  });
});