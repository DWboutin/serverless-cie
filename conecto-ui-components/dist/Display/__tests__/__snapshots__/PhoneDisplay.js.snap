// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneDisplay should match snapshot 1`] = `
<div
  className="to-phone-display"
>
  <PhoneShape />
  <div
    className="phone-wrapper"
  >
    <div
      className="wrapper"
    >
      <div
        className="header"
      >
        <div
          className="logo"
        >
          <Logo
            size={null}
            suffix={
              <FormattedMessage
                id="contractor"
                values={Object {}}
              />
            }
            type="roofing"
          />
        </div>
        <div
          className="button"
        >
          <Button
            comp="button"
            htmlType="button"
            type="bordered"
          >
            question@contect.ca
          </Button>
        </div>
      </div>
      <div
        className="content"
      >
        <div
          className="text-content"
        >
          test
        </div>
        <div
          className="bottom-svg"
        >
          <RoofingPhoneBg />
        </div>
      </div>
    </div>
  </div>
</div>
`;
